feat(helm): add plugin management commands

adamreese opened the pull request:
Add plugin management subcommands for installing and removing plugins
to `$HELM_HOST/plugins`.

Install accepts a vcs url or a local directory.

```
$ helm plugin install http://github.com/adamreese/helm-env
Installed plugin: env

$ helm plugin list
NAME    	VERSION	DESCRIPTION
env     	0.1.0  	Print out the helm environment.

$ helm plugin remove env
Removed plugin: env
```

closes #1977

technosophos commented on cmd/helm/plugin.go:
```diff
@@ -0,0 +1,72 @@
+	prog := exec.Command("sh", "-c", hook)
+	// TODO make this work on windows
+	// I think its ... ¯\_(ツ)_/¯
+	// prog := exec.Command("cmd", "/C", p.Metadata.Hooks.Install())
```
We should probably file an issue for windows work so we don't lose track of it.

technosophos commented:
My manual tests:

1. Start with Helm system with lots of existing plugins
2. Delete existing plugin

```console
⇒  helm plugin remove github
Error: remove /Users/mattbutcher/HelmHome/plugins/github: directory not empty
```

That is expected, since it was installed manually.

3. Manually delete the plugins
4. Re-add the plugin:

```console
⇒  helm plugin install https://github.com/technosophos/helm-gpg
Installed plugin: gpg
```

Repeated with a few other plugins. Each one tested manually.

5. List plugins

```console
⇒  helm plugin list
NAME      	VERSION	DESCRIPTION
env       	0.1.0  	Print the environment that the plugin was given, then exit.
example   	0.1.0  	This is a Helm plugin.
hello     	0.1.0  	This is a demonstration plugin that prints Hello and then exists.
github    	0.1.0  	This plugin provides GitHub services to Helm.
gpg       	0.1.0  	This plugin provides GnuPG services to Helm.
keybase   	0.1.0  	This plugin provides Keybase services to Helm.
last      	0.1.0  	Print the name of the last release.
template  	2.1.3+1	Render templates on the local client.
testgetter	0.1.0  	Print the environment that the plugin was given, then exit.

          	       	T...
tiller    	0.1.0  	Helpers for working with in-cluster tiller.
```

6. Delete a plugin

```console
⇒  helm plugin remove gpg
Removed plugin: gpg
```

7. List again

```console
⇒  helm plugin list
NAME      	VERSION	DESCRIPTION
env       	0.1.0  	Print the environment that the plugin was given, then exit.
example   	0.1.0  	This is a Helm plugin.
hello     	0.1.0  	This is a demonstration plugin that prints Hello and then exists.
github    	0.1.0  	This plugin provides GitHub services to Helm.
keybase   	0.1.0  	This plugin provides Keybase services to Helm.
last      	0.1.0  	Print the name of the last release.
template  	2.1.3+1	Render templates on the local client.
testgetter	0.1.0  	Print the environment that the plugin was given, then exit.

          	       	T...
tiller    	0.1.0  	Helpers for working with in-cluster tiller.
```

8. Install again

```console
⇒  helm plugin install https://github.com/technosophos/helm-gpg
Installed plugin: gpg
```

9. Try to install an upstream plugin into an existing directory that already has a plugin.

```console
⇒  helm plugin install https://github.com/adamreese/helm-last  
Error: symlink /Users/mattbutcher/HelmHome/cache/plugins/https-github.com-adamreese-helm-last /Users/mattbutcher/HelmHome/plugins/helm-last: file exists
```

That is as expected.

adamreese committed:
feat(helm): add plugin management commands

Add plugin management subcommands for installing and removing plugins
to `$HELM_HOST/plugins`.

Install accepts a vcs url or a local directory.

```
$ helm plugin install http://github.com/adamreese/helm-env
Installed plugin: env

$ helm plugin list
NAME    	VERSION	DESCRIPTION
env     	0.1.0  	Print out the helm environment.

$ helm plugin remove env
Removed plugin: env
```

closes #1977
